relay-server/src/endpoints/minidump.rs

```diff
@@ -97,7 +120,11 @@ async fn extract_multipart(
         minidump_item.set_payload(ContentType::Minidump, embedded);
     }
 
-    validate_minidump(&minidump_item.payload())?;
+    if validate_minidump(&minidump_item.payload()).is_err() {
```

Comment:
Small suggestion: Similar to how we check for embedded minidumps, could we invert the logic slightly to fully separate concerns? Right now, `validate_gzip_minidump` fails with an error "invalid minidump" where actually it tests for a gzip header. 

I'm thinking about control flow like this: 
1. Check for a compression header
    1. If there is compression, then try to decompress and propagate all errors
    2. Replace the payload bytes with the decompressed bytes
2. Now check the minidump header once.

Reply:
Yes, of course, this makes sense. I was only prepping a PoC to check for basic warning signs and whether you want automatic detection/decoding in the endpoint. I am also unhappy with the gzip being the fallback to the typical case 😅.